qualitas/exports/export.py

```diff
@@ -50,3 +56,158 @@ def check_issue_connection(zenhub_client, repo_id, issue_id):
             continue
 
     return False
+
+
+def get_org_releases(github_v4_client, org):
+    github_query = """\
+query {
+    organization(login: "%(org)s") {
+        repositories(first: 100
+                     orderBy: {
+                         direction: DESC
+                         field: PUSHED_AT
+                     }
+                     %(after)s) {
+            nodes {
+                name
+                url
+                pushedAt
+                refs(refPrefix: "refs/tags/"
+                     orderBy: {
+                         direction: ASC
+                         field: TAG_COMMIT_DATE
+                     }
+                     last: 1) {
+                    nodes {
+                        name
+                        target { commitUrl }
+                    }
+                }
+                defaultBranchRef {
+                    target {
+                        oid
+                        commitUrl
+                    }
+                }
+            }
+            pageInfo {
+                endCursor
+                hasNextPage
+            }
+        }
+    }
+}
+"""
+    after = ''
+    while True:
+        result = github_v4_client.call_api(
+            github_query % {'org': org, 'after': after}
+        )['data']['organization']['repositories']
+
+        for repo in result['nodes']:
+            release = repo['refs']['nodes']
+            head = repo['defaultBranchRef']
+            if release:
+                yield {
+                    'name': repo['name'],
+                    'url': repo['url'],
+                    'pushed_at': repo['pushedAt'],
+                    'release': {
+                        'version': release[0]['name'],
+                        'url': release[0]['target']['commitUrl'],
+                    },
+                    'head_ref': {
+                        'commit': head['target']['oid'],
+                        'url': head['target']['commitUrl'],
+                    },
+                }
+        after = 'after: "{}"'.format(result['pageInfo']['endCursor'])
+        if not result['pageInfo']['hasNextPage']:
+            break
+
+
+def parse_history_txt(server):
+    content = urlopen('https://{}/history.txt'.format(server)).read().decode(
+        'utf-8')
+    latest = re.split('---+\n', re.split('===+\n', content, maxsplit=1)[0])
+    release = {package: [version]
+               for package, version in json.loads(latest[0]).items()}
+    release_date = release.pop('date')[0]
+
+    if 'webview' in release and '(' in release['webview'][0]:
+        # webview version looks like:
+        #    fc509073b895ac03bc66a3191204c1f44c648175 (v0.50.0-11-gfc50907)
+        # only display the version in parentheses for webview
+        version = release['webview'][0].split('(')[1].split(')')[0]
```

Comment:
We have logic in cnx-automation as well to parse the history.txt file. I wonder if it's time for a pypi library to do this kind of work ;).